src/query/storage/prom_converter.go

```diff
@@ -202,10 +215,22 @@ func toPromConcurrently(
 	}
 
 	// Filter out empty series inplace.
+	meta := block.NewResultMetadata()
 	filteredList := seriesList[:0]
 	for _, s := range seriesList {
+		name := ""
+		for _, l := range s.Labels {
+			if string(l.Name) == model.MetricNameLabel {
```

Comment:
nit: may be worth trying to avoid the string conversion here too

Reply:
same re: promDefaultName, will follow up